src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -1,53 +1,57 @@
 package com.sprint.mission.discodeit.service.basic;
 
+import com.sprint.mission.discodeit.dto.CreateUserDTO;
+import com.sprint.mission.discodeit.dto.UpdateUserDTO;
+import com.sprint.mission.discodeit.dto.UserResponseDTO;
 import com.sprint.mission.discodeit.entity.User;
 import com.sprint.mission.discodeit.repository.UserRepository;
 import com.sprint.mission.discodeit.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
+@RequiredArgsConstructor
+@Service
 public class BasicUserService implements UserService {
     private final UserRepository userRepository;
 
-    // 생성자에서 UserRepository를 주입
-    public BasicUserService(UserRepository userRepository) {
-        this.userRepository = userRepository;
-    }
-
-    @Override
-    public void createUser(User user) {
-        userRepository.createUser(user); // Repository에 저장
-    }
-
-    // Basic*Service 구현체를 활용
     @Override
-    public User createUser(String userName, String email, String password) {
-        User user = new User(UUID.randomUUID(), System.currentTimeMillis(), System.currentTimeMillis(), userName, email, password);
-        userRepository.createUser(user); // Repository에 저장
-        return user;
+    public User create(CreateUserDTO createUserDTO) {
+        User user = new User(createUserDTO.getUsername(), createUserDTO.getPassword(), createUserDTO.getEmail());
+        return userRepository.save(user);
     }
 
     @Override
-    public User getUser(UUID id) {
-        return userRepository.getUser(id)
-                .orElseThrow(() -> new IllegalArgumentException("해당 ID의 사용자를 찾을 수 없습니다: " + id));
+    public UserResponseDTO find(UUID userId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new NoSuchElementException("User with id " + userId + " not found"));
+        return new UserResponseDTO(user);
     }
 
     @Override
-    public List<User> getAllUsers() {
-        return userRepository.getAllUsers();
+    public List<UserResponseDTO> findAll() {
+        return userRepository.findAll()
+                .stream()
+                .map(UserResponseDTO::new)
+                .collect(Collectors.toList());
```

Comment:
Stream.toList() 바로 사용해도 될 것 같네요.

```suggestion
                .toList();
```